algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java

```diff
@@ -119,21 +124,24 @@ public Set<ActionableEntry> create(@NotNull CkbEntry entry, @NotNull String sour
 
                 Set<String> curatedRelevantTreatmentApproaches = Sets.newHashSet(curator.isMatch(key));
 
-                actionableEntries.add(ImmutableActionableEntry.builder()
-                        .source(Knowledgebase.CKB_EVIDENCE)
-                        .sourceEvent(sourceEvent)
-                        .sourceUrls(sourceUrls)
-                        .treatment(ImmutableTreatment.builder()
-                                .name(treatment)
-                                .sourceRelevantTreatmentApproaches(sourceRelevantTreatmentApproaches)
-                                .relevantTreatmentApproaches(curatedRelevantTreatmentApproaches)
-                                .build())
-                        .applicableCancerType(cancerTypeExtraction.applicableCancerType())
-                        .blacklistCancerTypes(cancerTypeExtraction.blacklistedCancerTypes())
-                        .level(level)
-                        .direction(direction)
-                        .evidenceUrls(evidenceUrls)
-                        .build());
+                if (!blacklistEvidence.isBlacklistEvidence(treatment, cancerTypeExtraction.applicableCancerType().name(),
```

Comment:
It's fairly minor but you could consider evaluating for blacklisting at an earlier stage. 

E.g. at the moment you curate the treatment approach but that's only necessary if evidence is not blacklisted anyways

Reply:
correct, this can indeed in a earlier stage